Fix TypeScript 6.0 build failure: baseUrl deprecated

## Summary

PR #3481 (Renovate: TypeScript 5.9.3 → 6.0.2) CI build is failing.

## Error

```
tsconfig.build.json:3:2 - error TS5101: Option 'baseUrl' is deprecated and will stop functioning in TypeScript 7.0.
Specify compilerOption '"ignoreDeprecations": "6.0"' to silence this error.
```

Failed packages: `@markuplint/ml-ast`, `@markuplint/shared`, `@markuplint/i18n`, `@markuplint/test-tools`
(All packages are affected, but Nx parallel execution stops after the first 4 failures)

## Root Cause

The root `tsconfig.json` defines `"baseUrl": "."`, which is inherited by all packages' `tsconfig.build.json` through the extends chain. TypeScript 6.0 deprecated `baseUrl`, causing build errors.

## Proposed Fix

Remove `baseUrl` from the root `tsconfig.json` and rely on TS 6's default behavior (paths resolved relative to the tsconfig.json location).

## References

- https://github.com/d-zero-dev/tsconfig/issues/59
- PR #3481

--- a/package.json
+++ b/package.json
@@ -93,7 +93,7 @@
 		"typedoc": "0.28.17",
 		"typedoc-plugin-mdn-links": "5.1.1",
 		"typedoc-plugin-resolve-crossmodule-references": "0.3.3",
-		"typescript": "5.9.3",
+		"typescript": "6.0.2",
 		"vitest": "4.0.18"
 	},
 	"config": {

--- a/packages/@markuplint-dev/tsconfig/tsconfig.json
+++ b/packages/@markuplint-dev/tsconfig/tsconfig.json
@@ -1,5 +1,6 @@
 {
 	"compilerOptions": {
+		"types": ["node"],
 		"module": "NodeNext",
 		"target": "ES2022",
 		"strict": true,

--- a/packages/@markuplint/ml-core/src/ml-dom/node/document.ts
+++ b/packages/@markuplint/ml-core/src/ml-dom/node/document.ts
@@ -252,6 +252,17 @@ export class MLDocument<T extends RuleConfigValue, O extends PlainData = undefin
 	/**
 	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
 	 *
+	 * @deprecated
+	 * @unsupported
+	 * @implements DOM API: `Document`
+	 */
+	get activeViewTransition(): ViewTransition | null {
+		throw new UnexpectedCallError('Not supported "activeViewTransition" property');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
 	 * @unsupported
 	 * @implements DOM API: `Document`
 	 */
@@ -383,6 +394,17 @@ export class MLDocument<T extends RuleConfigValue, O extends PlainData = undefin
 	 */
 	get currentScript(): HTMLOrSVGScriptElement | null {
 		throw new UnexpectedCallError('Not supported "currentScript" property');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @deprecated
+	 * @unsupported
+	 * @implements DOM API: `Document`
+	 */
+	get customElementRegistry(): CustomElementRegistry | null {
+		throw new UnexpectedCallError('Not supported "customElementRegistry" property');
 	}
 
 	/**
@@ -955,6 +977,24 @@ export class MLDocument<T extends RuleConfigValue, O extends PlainData = undefin
 		  ) => any)
 		| null {
 		throw new UnexpectedCallError('Not supported "onclose" property');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @deprecated
+	 * @unsupported
+	 * @implements DOM API: `Document`
+	 */
+	get oncommand():
+		| ((
+				// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+				this: GlobalEventHandlers,
+				// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+				ev: Event,
+		  ) => any)
+		| null {
+		throw new UnexpectedCallError('Not supported "oncommand" property');
 	}
 
 	/**

--- a/packages/@markuplint/ml-core/src/ml-dom/node/element.ts
+++ b/packages/@markuplint/ml-core/src/ml-dom/node/element.ts
@@ -370,7 +370,7 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
-	get ariaControlsElements(): readonly MLElement<T, O>[] {
+	get ariaControlsElements(): readonly MLElement<T, O>[] | null {
 		throw new UnexpectedCallError('Not supported "ariaControlsElements" property');
 	}
 
@@ -392,7 +392,7 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
-	get ariaDescribedByElements(): readonly MLElement<T, O>[] {
+	get ariaDescribedByElements(): readonly MLElement<T, O>[] | null {
 		throw new UnexpectedCallError('Not supported "ariaDescribedByElements" property');
 	}
 
@@ -414,7 +414,7 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
-	get ariaDetailsElements(): readonly MLElement<T, O>[] {
+	get ariaDetailsElements(): readonly MLElement<T, O>[] | null {
 		throw new UnexpectedCallError('Not supported "ariaDetailsElements" property');
 	}
 
@@ -436,7 +436,7 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
-	get ariaErrorMessageElements(): readonly MLElement<T, O>[] {
+	get ariaErrorMessageElements(): readonly MLElement<T, O>[] | null {
 		throw new UnexpectedCallError('Not supported "ariaErrorMessageElements" property');
 	}
 
@@ -458,7 +458,7 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
-	get ariaFlowToElements(): readonly MLElement<T, O>[] {
+	get ariaFlowToElements(): readonly MLElement<T, O>[] | null {
 		throw new UnexpectedCallError('Not supported "ariaFlowToElements" property');
 	}
 
@@ -524,7 +524,7 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
-	get ariaLabelledByElements(): readonly MLElement<T, O>[] {
+	get ariaLabelledByElements(): readonly MLElement<T, O>[] | null {
 		throw new UnexpectedCallError('Not supported "ariaLabelledByElements" property');
 	}
 
@@ -601,7 +601,7 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
-	get ariaOwnsElements(): readonly MLElement<T, O>[] {
+	get ariaOwnsElements(): readonly MLElement<T, O>[] | null {
 		throw new UnexpectedCallError('Not supported "ariaOwnsElements" property');
 	}
 
@@ -981,6 +981,17 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
+	get customElementRegistry(): CustomElementRegistry | null {
+		throw new UnexpectedCallError('Not supported "customElementRegistry" property');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @deprecated
+	 * @unsupported
+	 * @implements DOM API: `Element`
+	 */
 	get dataset(): DOMStringMap {
 		throw new UnexpectedCallError('Not supported "dataset" property');
 	}
@@ -1179,7 +1190,7 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
-	get nonce(): string | undefined {
+	get nonce(): string {
 		throw new UnexpectedCallError('Not supported "nonce" property');
 	}
 
@@ -1524,6 +1535,24 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 		  ) => any)
 		| null {
 		throw new UnexpectedCallError('Not supported "onclose" property');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @deprecated
+	 * @unsupported
+	 * @implements DOM API: `Element`
+	 */
+	get oncommand():
+		| ((
+				// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+				this: GlobalEventHandlers,
+				// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+				ev: Event,
+		  ) => any)
+		| null {
+		throw new UnexpectedCallError('Not supported "oncommand" property');
 	}
 
 	/**
@@ -3838,6 +3867,11 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @implements DOM API: `Element`
 	 * @see https://dom.spec.whatwg.org/#ref-for-dom-element-matches%E2%91%A0
 	 */
+	matches<K extends keyof HTMLElementTagNameMap>(selectors: K): this is HTMLElementTagNameMap[K];
+	matches<K extends keyof SVGElementTagNameMap>(selectors: K): this is SVGElementTagNameMap[K];
+	matches<K extends keyof MathMLElementTagNameMap>(selectors: K): this is MathMLElementTagNameMap[K];
+	// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+	matches(selectors: string, scope?: MLParentNode<T, O>): boolean;
 	matches(
 		selector: string,
 		// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types

--- a/packages/@markuplint/ml-core/src/ml-dom/node/node.ts
+++ b/packages/@markuplint/ml-core/src/ml-dom/node/node.ts
@@ -890,6 +890,21 @@ export abstract class MLNode<
 	}
 
 	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @unsupported
+	 * @implements DOM API: `Node`
+	 */
+	moveBefore(
+		// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+		node: Node,
+		// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+		child: Node | null,
+	): Node {
+		throw new UnexpectedCallError('Not supported "moveBefore" method');
+	}
+
+	/**
 	 * @implements `@markuplint/ml-core` API: `MLNode`
 	 */
 	is<NType extends NodeType>(nodeType: NType): this is NodeTypeOf<NType, T, O> {

--- a/packages/@markuplint/test-tools/src/index.ts
+++ b/packages/@markuplint/test-tools/src/index.ts
@@ -45,7 +45,6 @@ export function createJSDOMElement(
 		globalThis.Element = dom.window.Element;
 		globalThis.getComputedStyle = dom.window.getComputedStyle.bind(dom.window);
 		if (matches) {
-			 
 			dom.window.Element.prototype.matches = function (this: Element, selector: string) {
 				return matches.call(this, selector);
 			} as any;
@@ -58,7 +57,6 @@ export function createJSDOMElement(
 	globalThis.Element = dom.window.Element;
 	globalThis.getComputedStyle = dom.window.getComputedStyle.bind(dom.window);
 	if (matches) {
-		 
 		dom.window.Element.prototype.matches = function (this: Element, selector: string) {
 			return matches.call(this, selector);
 		} as any;

--- a/packages/@markuplint/test-tools/src/index.ts
+++ b/packages/@markuplint/test-tools/src/index.ts
@@ -45,9 +45,10 @@ export function createJSDOMElement(
 		globalThis.Element = dom.window.Element;
 		globalThis.getComputedStyle = dom.window.getComputedStyle.bind(dom.window);
 		if (matches) {
-			dom.window.Element.prototype.matches = function (selector: string) {
+			 
+			dom.window.Element.prototype.matches = function (this: Element, selector: string) {
 				return matches.call(this, selector);
-			};
+			} as any;
 		}
 		return dom.window.document.querySelector('html') as Element;
 	}
@@ -57,9 +58,10 @@ export function createJSDOMElement(
 	globalThis.Element = dom.window.Element;
 	globalThis.getComputedStyle = dom.window.getComputedStyle.bind(dom.window);
 	if (matches) {
-		dom.window.Element.prototype.matches = function (selector: string) {
+		 
+		dom.window.Element.prototype.matches = function (this: Element, selector: string) {
 			return matches.call(this, selector);
-		};
+		} as any;
 	}
 
 	const containerTag = html.indexOf('<td') === 0 ? 'tr' : 'div';

--- a/test/isolated-env/ts/tsconfig.json
+++ b/test/isolated-env/ts/tsconfig.json
@@ -1,5 +1,6 @@
 {
 	"compilerOptions": {
+		"types": ["node"],
 		"target": "ESNext",
 		"module": "NodeNext",
 		"moduleResolution": "NodeNext"

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,10 +1,3 @@
 {
-	"extends": ["@markuplint-dev/tsconfig"],
-	"compilerOptions": {
-		"baseUrl": ".",
-		"paths": {
-			"@markuplint/*": ["./packages/@markuplint/*/src"],
-			"*": ["./packages/*/src"]
-		}
-	}
+	"extends": ["@markuplint-dev/tsconfig"]
 }

